src/platform_impl/orbital/event_loop.rs

```diff
@@ -333,39 +333,50 @@ impl<T: 'static> EventLoop<T> {
     {
         match event_option {
             EventOption::Key(KeyEvent {
-                character: _,
+                character,
                 scancode,
                 pressed,
             }) => {
-                if scancode != 0 {
-                    let physical_key = convert_scancode(scancode);
-                    let modifiers_before = event_state.keyboard;
-                    event_state.key(physical_key, pressed);
-                    event_handler(event::Event::WindowEvent {
-                        window_id: RootWindowId(window_id),
-                        event: event::WindowEvent::KeyboardInput {
-                            device_id: event::DeviceId(DeviceId),
-                            event: event::KeyEvent {
-                                logical_key: Key::Unidentified(NativeKey::Unidentified),
-                                physical_key,
-                                location: KeyLocation::Standard,
-                                state: element_state(pressed),
-                                repeat: false,
-                                text: None,
-
-                                platform_specific: KeyEventExtra {},
-                            },
-                            is_synthetic: false,
+                let physical_key = convert_scancode(scancode);
+                let (logical_key, text) = if character == '\0' {
+                    (Key::Unidentified(NativeKey::Unidentified), None)
+                } else {
+                    let mut character_buf = [0u8; 4];
+                    let character_str = character.encode_utf8(&mut character_buf);
+                    (
+                        Key::Character(character_str.into()),
+                        if pressed {
+                            Some(character_str.into())
+                        } else {
+                            None
                         },
-                    });
+                    )
+                };
+                let modifiers_before = event_state.keyboard;
+                event_state.key(physical_key, pressed);
+                event_handler(event::Event::WindowEvent {
+                    window_id: RootWindowId(window_id),
+                    event: event::WindowEvent::KeyboardInput {
+                        device_id: event::DeviceId(DeviceId),
+                        event: event::KeyEvent {
+                            logical_key,
+                            physical_key,
+                            location: KeyLocation::Standard,
+                            state: element_state(pressed),
+                            repeat: false,
+                            text,
+                            platform_specific: KeyEventExtra {},
```

Comment:
if it's not implemented you can `Default::default`?

Reply:
Perhaps, unless it is considered a requirement to support text_with_all_modifiers and key_without_modifiers correctly before this is merged. Then, I will use this to pass those around like the Linux implementation does.

Reply:
I'm now using this like the Linux platform does.